### PR TITLE
Updates to README and Rule Authoring guide

### PR DESCRIPTION
**Pull request by JohnathonMohr**
- General wording updates
- Clarify .NET 5 requirement
- Add instructions for how to download (GitHub releases)
- Instructions for executing the analyzer on the CLI
  - Clarification on where the built-in rules come from
- Put note about using parameters in analysis on its own line
- Replace references to `Evaluate` operator with `allOf` in rule authoring guide.
  - Add note that while `Evaluate` isn't yet available, `allOf` is a sufficient work-around.

### CODE REVIEW

**Review by yane3628 — CHANGES_REQUESTED** [review]
Can you also add more details about the CLI to close out this issue https://github.com/Azure/template-analyzer/issues/34

File: docs/authoring-json-rules.md [hunk 1]
```diff
@@ -313,6 +313,8 @@ Wraps a single `Evaluation`.  The result of the operator is exactly the result o
 
 This operator is most commonly useful in combination with a [`where` condition](#where-conditions), where `resourceType` or `path` may need to be [scoped down](#scopes) multiple times.
 
+**NOTE**: `evaluate` is not yet supported.  As a workaround, replace it with `allOf` or `anyOf` containing a single `Evaluation`.
```
[the remark this answers is not in the document]
Comment (VeraBE) [reply docs/authoring-json-rules.md:323]: "replace it with an..."?

**Review by VeraBE — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -2,32 +2,48 @@
...
+The Template BPA depends on a couple inputs – an ARM template to analyze, specified with the `-t` or `--template-file-path` argument, and an (optional) ARM template parameters file, specified with the `-p` or `--parameters-file-path` argument. The Template BPA then runs the [configured rules](#understanding-and-customizing-rules) against the provided ARM template and its corresponding [template parameters](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/parameter-files), if specified. If no template parameters are specified, then the Template BPA generates the minimum number of placeholder parameters to properly evaluate [template functions](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions) in the ARM template.
```
Comment (yane3628) [inline README.md:24]: Saying it "depends on" and then noting that parameters is optional seems strange to me. Instead maybe do something like 

The Template BPA allows for the following inputs:
- `-t` or `--template-file-path` - an ARM template to analyze
- (Optional) `-p` or `--parameters-file-path` - an ARM template parameters file

**Review by yane3628 — CHANGES_REQUESTED** [review]

**Review by yane3628 — APPROVED** [review]
Just because you put the inputs in a table format, maybe instead of "(Optional)" we add a required column? Similar to the [templates documentation](https://docs.microsoft.com/en-us/azure/templates/microsoft.containerservice/managedclusters?tabs=json#microsoftcontainerservicemanagedclusters-object).

**Comment by JohnathonMohr** [comment]
> Just because you put the inputs in a table format, maybe instead of "(Optional)" we add a required column? Similar to the [templates documentation](https://docs.microsoft.com/en-us/azure/templates/microsoft.containerservice/managedclusters?tabs=json#microsoftcontainerservicemanagedclusters-object).

I think because there are only 2 inputs, having a dedicated column for that might be a little too much.

**Comment by yane3628** [comment]
@JohnathonMohr Can you link this PR to this issue? https://github.com/Azure/template-analyzer/issues/34 You'll likely have to update the status of the issue since the PR was closed before it was linked.

**Pull request merged**